public/js/graph/react_graph.js

```diff
@@ -303,11 +306,41 @@ var Graph = React.createClass({
         );
     },
 
+    incrementZoom: function(increase) {
+        if (increase) {
+            if (this.state.zoomFactor > 0.5) {
+                this.setState({zoomFactor: this.state.zoomFactor - 0.05});
+            }
+        } else {
+            if (this.state.zoomFactor < 1.5) {
+                this.setState({zoomFactor: this.state.zoomFactor + 0.05});
+            }
+        }
+
+    },
+
     render: function () {
         // not all of these properties are supported in React
-        var svgAttrs = {width: this.props.width, height: this.props.height};
+        var svgAttrs = {
+            width: '100%',
+            height: '100%',
+            viewBox: '0 0 ' +
+                (this.props.width * this.state.zoomFactor) + ' ' + (this.props.height * this.state.zoomFactor),
+            preserveAspectRatio: 'xMinYMin'
+        };
 
         return (
+            <div>
+            <Button
+                id='zoom-in-button'
+                sourceImg="static/res/ico/in.png"
+                mouseDown={this.incrementZoom}
+                mouseDownArg={true}/>
```

Comment:
You usually don't want to pass in a separate prop for an argument. You can change the mouseDown to be an anonymous function (`() => ...`) whose body calls `this.incrementZoom` on the correct argument.